sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ExceptionUtil.java

```diff
@@ -78,8 +77,9 @@ public static Exception toException(String errorCondition, String description, A
             case NOT_FOUND:
                 return distinguishNotFound(description, errorContext);
             default:
-                throw new IllegalArgumentException(String.format(Locale.ROOT, "This condition '%s' is not known.",
-                    condition));
+                return new AmqpException(false, condition, String.format("errorCondition[%s]. description[%s] "
+                        + "Condition could not be mapped to a transient condition or not.",
```

Comment:
```suggestion
                        + "Condition could not be mapped to a transient condition.",
```

Not sure what the `or not` was adding to the error message